go/libraries/doltcore/schema/typeinfo/json.go

```diff
@@ -0,0 +1,200 @@
+	if val, ok := v.(types.JSON); ok {
+		return json.NomsJSON(val), nil
+	}
+	if _, ok := v.(types.Null); ok || v == nil {
+		return nil, nil
+	}
+	return nil, fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), v.Kind())
+}
+
+// ReadFrom reads a go value from a noms types.CodecReader directly
+func (ti *jsonType) ReadFrom(_ *types.NomsBinFormat, reader types.CodecReader) (interface{}, error) {
+	k := reader.PeekKind()
+	switch k {
+	case types.JSONKind:
+		js, err := reader.ReadJSON()
+		if err != nil {
+			return nil, err
+		}
+		return json.NomsJSON(js), nil
+	case types.NullKind:
+		_ = reader.ReadKind()
+		return nil, nil
+	}
+
+	return nil, fmt.Errorf(`"%v" cannot convert NomsKind "%v" to a value`, ti.String(), k)
+}
+
+// ConvertValueToNomsValue implements TypeInfo interface.
+func (ti *jsonType) ConvertValueToNomsValue(ctx context.Context, vrw types.ValueReadWriter, v interface{}) (types.Value, error) {
+	if v == nil {
+		return types.NullValue, nil
+	}
+
+	jsDoc, err := ti.jsonType.Convert(v)
+	if err != nil {
+		return nil, err
+	}
+
+	jsVal, ok := jsDoc.(sql.JSONValue)
+	if !ok {
+		return nil, fmt.Errorf(`"%v" cannot convert value "%v" of type "%T" as it is invalid`, ti.String(), v, v)
+	}
+
+	noms, err := json.NomsJSONFromJSONValue(ctx, vrw, jsVal)
+	if err != nil {
+		return nil, err
+	}
+
+	return types.JSON(noms), err
+}
+
+// Equals implements TypeInfo interface.
+func (ti *jsonType) Equals(other TypeInfo) bool {
+	return ti.GetTypeIdentifier() == other.GetTypeIdentifier()
+}
+
+// FormatValue implements TypeInfo interface.
+func (ti *jsonType) FormatValue(v types.Value) (*string, error) {
+	if _, ok := v.(types.Null); ok || v == nil {
+		return nil, nil
+	}
+	if noms, ok := v.(types.JSON); ok {
+		// TODO(andy) fix context
+		s, err := json.NomsJSON(noms).ToString(sql.NewEmptyContext())
+		if err != nil {
+			return nil, err
+		}
+		return &s, nil
+	}
+	return nil, fmt.Errorf(`"%v" has unexpectedly encountered a value of type "%T" from embedded type`, ti.String(), v)
+}
+
+// GetTypeIdentifier implements TypeInfo interface.
+func (ti *jsonType) GetTypeIdentifier() Identifier {
+	return JSONTypeIdentifier
+}
+
+// GetTypeParams implements TypeInfo interface.
+func (ti *jsonType) GetTypeParams() map[string]string {
+	return nil
+}
+
+// IsValid implements TypeInfo interface.
+func (ti *jsonType) IsValid(v types.Value) bool {
+	if v == nil {
+		return true
+	}
+	switch v.(type) {
+	case types.JSON:
+		return true
+	case types.Null:
+		return true
+	default:
+		return false
+	}
+}
+
+// NomsKind implements TypeInfo interface.
+func (ti *jsonType) NomsKind() types.NomsKind {
+	return types.JSONKind
+}
+
+// ParseValue implements TypeInfo interface.
+func (ti *jsonType) ParseValue(ctx context.Context, vrw types.ValueReadWriter, str *string) (types.Value, error) {
+	if str == nil {
+		return types.NullValue, nil
+	}
+	// TODO(andy) 2 step conversion here gonna be slow
```

Comment:
Also planning to remove this function. But in the meantime, you should use the `ctx` parameter.